docs/rfcs/1005-realm-key-rotation.md

```diff
@@ -0,0 +1,821 @@
+removing realm vlobs (cannot remove a realm given it is related to certificates) or
+organization form the database.
+
+### 5.3 Server data migration
+
+Drop tables:
+
+- `vlob_encryption_revision`
+- `message`
+
+Table `vlob_atom`:
+
+- Remove all rows that contains old encryption revision data (i.e. the table only contains
+  the vlob currently accessible by the users).
+- Remove column `vlob_encryption_revision`
+- Add column `key_index` (`INTEGER NOT NULL`). This defaults to 1 when migrating data.
+
+Table `realm`
+
+- Remove columns `encryption_revision`, `maintenance_started_by`, `maintenance_started_on` and `maintenance_type`
+
+## 6 Changes: Client-side
+
+### 6.1 Certificate Ops
+
+`RealmKeyRotationCertificate` should be checked (only OWNER of the realm can create such
+certificate).
+
+### 6.2 Workspaces listing
+
+The client has to rely on both user manifest and certificates for this:
+
+1) Entries present in the user manifest with a corresponding realm role certificate
+   (with non-null role): nominal case.
+2) Entries only present in the user manifest: corresponds to a newly created
+   workspaces that have not yet been synced with the server
+3) No entry in the user manifest, but a realm role certificate (with non-null role)
+   exists: corresponds to a newly shared realm
+
+Case 1 and 2 work just fine: the workspace name is present in the user manifest.
+
+Case 3 requires to go fetch the workspace name from the `KeysBundleAccess` that
+correspond to the key index when the realm was shared with our user.
+
+> **TODO:**
+>
+> It's most likely last sentence made you scream internally considering the complexity
+> of the operation...
+>
+> The alternatives are:
+>
+> - always provide the workspace label in the `KeysBundleAccess` (and not only for
+>   `KeysBundleAccess` provided while calling `realm_share`). The downside is this is
+>   wasteful in server database storage.
+> - encrypt the workspace label with the recipient public key and put it in the realm role
+>   certificate. It is also wasteful, both for the server and the client that store every
+>   certificate.
+> - pass the workspace label as a separated encrypted (with recipient public key) field
+>   in `realm_share` command, then provide a `realm_get_workspace_label` command to
+>   retrieve it. This add very ad-hoc APIs that will most likely be deprecated when we
+>   will introduce workspace label controlled by workspace owners.
+
+Case 3 should be handle by the user manager:
+
+- On init the user manager compare the user manifest with the realm roles certificates
+  to detect realm with missing entry
+- The user manifest also listen for new certificates event to re-do the check
+- The user manifest then fill the missing entries using `realm_get_keys_bundle` server command
+- When a workspace listing is done, the missing entries are simply ignored ()
+
+### 6.3 Fetching keys-bundle
```

Comment:
I'm not confident about having to fetch the bundle each time (given it simplify not having to handle a cache for it). 
That require to do the keys check every time we start parsec and fells like `rust-analyzer` cold start when I start my IDE

Reply:
At first I was thinking that `Fetching keys-bundle` was made at the start of the application, but will likely be done when we need to upload/download some data in that case that would add latency on the user for the first sync :thinking:

Reply:
The fetch keys bundle is only needed when uploading/downloading data to/from the server.
Hence when we do the fetch keys bundle we were about to do a server request no matter what, so having two request instead of one the first time it no big deal (adding a in-ram cache on this is totally trivial).